src/include/mutex.i

```diff
@@ -294,20 +301,24 @@ static inline void
 __wt_spin_lock_track(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
 {
 	uint64_t time_start, time_stop;
-	int64_t **stats;
+	int64_t *session_stats, **stats;
 
 	if (t->stat_count_off != -1 && WT_STAT_ENABLED(session)) {
 		time_start = __wt_clock(session);
 		__wt_spin_lock(session, t);
 		time_stop = __wt_clock(session);
 		stats = (int64_t **)S2C(session)->stats;
+		session_stats = (int64_t *)&(session->stats);
 		stats[session->stat_bucket][t->stat_count_off]++;
 		if (F_ISSET(session, WT_SESSION_INTERNAL))
 			stats[session->stat_bucket][t->stat_int_usecs_off] +=
 			    (int64_t)WT_CLOCKDIFF_US(time_stop, time_start);
-		else
+		else {
 			stats[session->stat_bucket][t->stat_app_usecs_off] +=
 			    (int64_t)WT_CLOCKDIFF_US(time_stop, time_start);
+			session_stats[t->stat_session_usecs_off] +=
```

Comment:
I think it is better to track these statistics for internal sessions as well as other statistics we are not differentiating. This will also help us in diagnosis of some scenario's.